MACOS.md

```diff
@@ -198,14 +184,63 @@ brew upgrade openssl     || brew install openssl
 ```
 
 
-## Sublime Text 3 - Your text editor
+## Visual Studio Code
+
+### Installation
+
+1. Download [Visual Studio Code for macOS](https://code.visualstudio.com/Download).
+2. Open the browser's download list and locate the downloaded file.
+3. Select the `magnifying glass` icon to open the file in `Finder`.
+4. Drag `Visual Studio Code.app` to the **Applications** folder, making it available in the macOS Launchpad. **Don't skip this step!**
+5. Add VS Code to your Dock by right-clicking on the icon to bring up the context menu and choosing Options, Keep in Dock.
+
+### Launching from the terminal
+
+You can also run VS Code from the terminal by typing `code` after adding it to the path.
 
-A text editor is one of the most important tools of a developer. Go to [this page](http://www.sublimetext.com/3) and download **Sublime Text 3** for OS X. Install it (double click the downloaded file and drag & drop the app **into** the `Applications` folder, **don't skip this**). If you had Sublime Text 2 installed before, please uninstall it (by dragging/dropping it to the Trash).
+Launch VS Code by clicking the icon in your Dock.
 
-Sublime Text is free without any time limitation but a popup will appear every ten saves to remind you there is a license to buy. You can hit `Esc` when this happens, but feel free to buy Sublime Text if you really like this one (there are alternatives).
+Open the Command Palette (`Cmd` + `Shift` + `P`) and type `shell command` to find the Shell Command: Install 'code' command in PATH command:
 
-Again, make sure that Sublime Text is there, not in the disk image you downloaded. To make sure it's correct, once Sublime Text is installed, unmount the "Sublime Text 3" disk in the left panel of Finder. Finder will complain if something went wrong. Ask a teacher.
+![](images/mac_vscode_command.png)
 
+Now quit the Terminal (`⌘` + `Q`) and restart it.
+
+Try typing `code` :point_right: if VS Code opens in new window, you can proceed to the next point!
+
+
+## VS Code Extensions
+
+### Installation
+
+Let's install some useful extensions to VS Code:
+
+**Extensions**
+- [Sublime Text Keymap and Settings Importer](https://marketplace.visualstudio.com/items?itemName=ms-vscode.sublime-keybindings)
+- [Live Share](https://marketplace.visualstudio.com/items?itemName=MS-vsliveshare.vsliveshare)
+- [ruby-rubocop](https://marketplace.visualstudio.com/items?itemName=misogi.ruby-rubocop)
+- [Ruby on Rails](https://marketplace.visualstudio.com/items?itemName=hridoy.rails-snippets)
+- [ESLint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint)
+
+Copy the command below then paste them in your terminal:
```

Comment:
It's better if you put this paragraph _on top_ of the list of Extensions. That way students will run the command, and then you can say "OK while this command is running, know that we are installing the following:", see my point?

Reply:
:+1: :pray: